index.html

```diff
@@ -36,98 +36,95 @@ <h2>Brigade is a tool for running scriptable, automated tasks in the cloud &mdas
-const { events, Job , Group} = require("brigadier");
-const dest = "$GOPATH/src/github.com/technosophos/ulid";
-
-events.on("push", (e, p) => {
-  console.log(e.payload)
-  var gh = JSON.parse(e.payload)
-  var test = new Job("test", "golang:1.9")
-  test.tasks = [
-    "mkdir -p " + dest,
-    "cp -a /src/* " + dest,
-    "cd " + dest,
-    "go get -u github.com/golang/dep/cmd/dep",
-    "dep ensure",
-    "make test"
-  ];
-  test.run()
-});
+<em>// Run unit tests for a Github push using TypeScript</em>
+import { events, Event, Job } from "@brigadecore/brigadier"
+
+const localPath = "/workspaces/brigade"
+
+events.on("brigade.sh/github", "push", async event => {
+  let test = new Job("test", "debian:latest", event)
+  test.primaryContainer.sourceMountPath = localPath
+  test.primaryContainer.workingDirectory = localPath
+  test.primaryContainer.command = ["bash"]
+  test.primaryContainer.arguments = ["-c", "make test"]
+  await test.run()
+})
+
+events.process()
               </pre>
             </li>
             <li data-orbit-slide="headline-2">
               <pre>
-<em>// Updating a cosmosDB database</em>
-const { events, Job } = require("brigadier")
-
-events.on("exec", (e, p) => {
-  var mongo = new Job("update-db", "mongo:3.2")
-  mongo.storage.enabled = false
-  mongo.tasks = [
-      dbCmd(p, `db.mydb.insert(${e.payload})`)
-  ]
-  mongo.run()
-})
+<em>// Update a CosmosDB database using JavaScript</em>
+const { events, Job } = require("@brigadecore/brigadier");
+
+events.on("brigade.sh/cli", "exec", async event => {
+  let mongo = new Job("update-db", "mongo:3.2", event);
+  mongo.primaryContainer.environment = {
+    COSMOS_NAME: event.project.secrets.cosmosName,
+    COSMOS_KEY: event.project.secrets.cosmosKey
+  };
+  mongo.primaryContainer.command = ["bash"];
+  mongo.primaryContainer.arguments = [
+    "-c",
+    dbCmd(event.project, `db.mydb.insert(${event.payload})`)
+  ];
+  await mongo.run();
+});
 
 function dbCmd(p, script) {
-  return `mongo ${p.secrets.cosmosName}.documents.azure.com:10255/test ` +
-    `-u ${p.secrets.cosmosName} -p  ${p.secrets.cosmosKey} --ssl --sslAllowInvalidCertificates ` +
+  return 'mongo ${COSMOS_NAME}.documents.azure.com:10255/test ' +
+    '-u ${COSMOS_NAME} -p ${COSMOS_KEY} --ssl --sslAllowInvalidCertificates ' +
     `--eval '${script}'`
 }
+
+events.process();
               </pre>
             </li>
             <li data-orbit-slide="headline-3">
               <pre>
-<em>// Sending a Slack message</em>
-const { events, Job } = require("brigadier")
+<em>// Send a Slack message in JavaScript</em>
+const { events, Job } = require("@brigadecore/brigadier");
 
-events.on("exec", (e, p) => {
-  var slack = new Job("slack-notify", "technosophos/slack-notify:latest", ["/slack-notify"])
-  slack.env = {
-    SLACK_WEBHOOK: p.secrets.SLACK_WEBHOOK,
+events.on("brigade.sh/cli", "exec", async event => {
+  let slack = new Job("slack-notify", "technosophos/slack-notify:latest", event);
```

Comment:
❤️  @technosophos of course, but I'd have a preference for examples involving actively maintained things.

Reply:
😢 Also fair.  Wanna suggest a replacement example?

Reply:
Thinking...

Reply:
@krancour I updated this example to use the forthcoming Slack gateway: https://github.com/brigadecore/brigade-www/pull/49/commits/c2c1d3aa5e06826bf871138c0e0f6979efc39c58 WDYT?